"""
Задание 6. В программе генерируется случайное целое число от 0 до 100.
Пользователь должен его отгадать не более чем за 10 попыток. После каждой
неудачной попытки должно сообщаться больше или меньше введенное пользователем
число, чем то, что загадано. Если за 10 попыток число не отгадано,
то вывести загаданное число.
Решите через рекурсию. В задании нельзя применять циклы.
"""

import random


def random_num(x=101, num=0, attempt=11):
    """
    Данная функция random_num загадывает число от 0 до 100 и приглашает
    пользователя отгадеть его с 10 попыток.
    В данной функции участвуют три переменных: x - хранит загаданное число,
    num - хранит число пользователя, attempt - хранит число попыток.
    """
    if x > 100:
        print('Отгадайте число!')
        x = random.randint(0, 100)
        random_num(x=x, num=0, attempt=11)
    else:
        attempt -= 1
        if attempt == 0:
            return print(f'У Вас закончились попытки!!! Загаданное число: {x}')
        try:
            num = int(input(f'Осталось попыток - {attempt}! '
                            f'Введите число от 0 до 100: '))
            if num < 0 or num > 100:
                print('Введено некорректное значение! Повторите попытку!')
                random_num(x=x, num=0, attempt=attempt)
        except ValueError:
            print('Введено некорректное значение! Повторите попытку!')
            random_num(x=x, num=0, attempt=attempt)
        if num == x:
            print('Вы угадали!!!')
            return
        else:
            if num > x:
                print('Ваше число больше, чем загаданное.')
                random_num(x=x, num=0, attempt=attempt)
            else:
                print('Ваше число меньше, чем загаданное.')
                random_num(x=x, num=0, attempt=attempt)


random_num()
